### ISSUE TITLE
Error while building with MSYS2 on Windows

### ISSUE THREAD

**Issue by krishean** [snesrev/zelda3#151]
### Describe your bug here. And how to reproduce it.

When building on Windows with MSYS2 under the MinGW64 or MinGW32 environments with `make` I get the following error:

```
cc -c -O2 -Werror -IC:/msys64/mingw64/include/SDL2 -Dmain=SDL_main -DSYSTEM_VOLUME_MIXER_AVAILABLE=0 glsl_shader.c -o glsl_shader.o
In file included from glsl_shader.h:4,
                 from glsl_shader.c:3:
types.h:37: error: "FORCEINLINE" redefined [-Werror]
   37 | #define FORCEINLINE inline
      |
In file included from C:/msys64/mingw64/include/minwindef.h:163,
                 from C:/msys64/mingw64/include/windef.h:9,
                 from C:/msys64/mingw64/include/windows.h:69,
                 from third_party/gl_core/gl_core_3_1.h:37,
                 from glsl_shader.c:2:
C:/msys64/mingw64/include/winnt.h:248: note: this is the location of the previous definition
  248 | #define FORCEINLINE __forceinline
      |
cc1.exe: all warnings being treated as errors
make: *** [Makefile:17: glsl_shader.o] Error 1
```

A similar error also occurs when cross compiling on Linux with MinGW via `CC=x86_64-w64-mingw32-gcc make` or `CC=i686-w64-mingw32-gcc make`:

```
x86_64-w64-mingw32-gcc -c -O2 -Werror -I../SDL2-2.24.1/x86_64-w64-mingw32/include/SDL2 -Dmain=SDL_main -DSYSTEM_VOLUME_MIXER_AVAILABLE=0 opengl.c -o opengl.o
In file included from opengl.c:5:0:
types.h:37:0: error: "FORCEINLINE" redefined [-Werror]
 #define FORCEINLINE inline
 
In file included from /usr/share/mingw-w64/include/minwindef.h:163:0,
                 from /usr/share/mingw-w64/include/windef.h:8,
                 from /usr/share/mingw-w64/include/windows.h:69,
                 from third_party/gl_core/gl_core_3_1.h:37,
                 from opengl.c:1:
/usr/share/mingw-w64/include/winnt.h:231:0: note: this is the location of the previous definition
 #define FORCEINLINE __forceinline
 
cc1: all warnings being treated as errors
Makefile:17: recipe for target 'opengl.o' failed
make: *** [opengl.o] Error 1
```

### What is your build target?

Windows

**Comment by snesrev** [comment]
Can you check if this is fixed in bf30ca05480103c118ef4503f2884cff9334e6dc

**Comment by krishean** [comment]
Successfully builds with MSYS2 and cross compiles on Linux as of bf30ca0
